python/common/org/python/types/Type.java

```diff
@@ -465,6 +465,9 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                             }
                             kwargs.put(arg_name, args[a]);
                         }
+                    } else {
+                        // More arguments than expected, call the constructor anyway to handle too many arguments
+                        adjusted_args = args;
```

Comment:
I thought about throwing a TypeError here, but the specific constructor has to do it because the error messages are not consistent across datatypes.